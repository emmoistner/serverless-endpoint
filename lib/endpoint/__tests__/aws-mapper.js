'use strict';

/* global describe jest test expect */
jest.dontMock('../mappers/aws-mapper');
jest.dontMock('../../util/is-json');

const awsMapper = require('../mappers/aws-mapper');
const eventObject = require('./api-gateway-event.json');
const Req = require('../req');
const Res = require('../res');

describe('mappers.aws-mapper.js', () => {
  test('convertEventToRequest should map event to an object res can consume', () => {
    expect(awsMapper.convertEventToRequest(eventObject)).toMatchSnapshot();
  });

  test('convertEventToRequest should apply sensible defaults for empty event object', () => {
    expect(awsMapper.convertEventToRequest()).toMatchSnapshot();
  });

  test('convertContextToResponse should map event to an object req can consume', () => {
    const context = { succeed: function succeed() {}, fail: function fail() {} };
    const preResObject = awsMapper.convertContextToResponse(context);
    expect(preResObject.done).toBeInstanceOf(Function);
    expect(preResObject.error).toBeInstanceOf(Function);
  });

  test('convertContextToResponse should map event to an object req can consume', () => {
    const context = { succeed: function succeed() {}, fail: function fail() {} };
    const event = eventObject;
    const mappedObject = awsMapper.apply({}, [event, context]);
    expect(mappedObject.req).toBeInstanceOf(Req);
    expect(mappedObject.res).toBeInstanceOf(Res);
  });

  test('calling awsMapper with apply can receive options', () => {
    const context = { succeed: function succeed() {}, fail: function fail() {} };
    const event = eventObject;
    const message = awsMapper.convertContextToResponse(context);
    const configOptions = { options: {} };

    awsMapper.apply(configOptions, [event, context]);

    expect(Res).toBeCalledWith(message, configOptions.options);
  });

  test('convertBody should handle stringified json', () => {
    // eslint-disable-next-line quotes
    expect(awsMapper.convertBody("{\"name\": \"Dave\"}")).toMatchSnapshot();
  });

  test('convertBody should handle object', () => {
    // eslint-disable-next-line quotes
    expect(awsMapper.convertBody({ name: 'Dave' })).toEqual({ name: 'Dave' });
  });

  test('convertBody should handle undefined', () => {
    // eslint-disable-next-line quotes
    expect(awsMapper.convertBody()).toEqual({});
  });
});
