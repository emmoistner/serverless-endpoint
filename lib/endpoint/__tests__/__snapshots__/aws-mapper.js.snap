// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mappers.aws-mapper.js convertBody should handle stringified json 1`] = `
Object {
  "name": "Dave",
}
`;

exports[`mappers.aws-mapper.js convertEventToRequest should apply sensible defaults for empty event object 1`] = `
Object {
  "apiId": undefined,
  "authorizer": Object {},
  "body": Object {},
  "headers": Object {},
  "id": undefined,
  "identity": Object {},
  "method": undefined,
  "og": Object {
    "context": undefined,
    "event": Object {},
  },
  "params": Object {},
  "provider": "aws",
  "query": Object {},
  "resource": undefined,
  "stage": undefined,
}
`;

exports[`mappers.aws-mapper.js convertEventToRequest should map event to an object res can consume 1`] = `
Object {
  "apiId": "xxxxxxxxxx",
  "authorizer": Object {},
  "body": Object {
    "name": "Ethan",
  },
  "headers": Object {
    "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8",
    "Accept-Encoding": "gzip, deflate, sdch, br",
    "Accept-Language": "en-US,en;q=0.8",
    "CloudFront-Forwarded-Proto": "https",
    "CloudFront-Is-Desktop-Viewer": "true",
    "CloudFront-Is-Mobile-Viewer": "false",
    "CloudFront-Is-SmartTV-Viewer": "false",
    "CloudFront-Is-Tablet-Viewer": "false",
    "CloudFront-Viewer-Country": "US",
    "Content-Type": "application/json",
    "Host": "xxxxxxxxxx.execute-api.us-east-1.amazonaws.com",
    "Upgrade-Insecure-Requests": "1",
    "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36",
    "Via": "1.1 xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.cloudfront.net (CloudFront)",
    "X-Amz-Cf-Id": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx_xxxxxxxxxxx_xxxx==",
    "X-Forwarded-For": "11.111.111.111, 11.111.111.111",
    "X-Forwarded-Port": "111",
    "X-Forwarded-Proto": "http",
  },
  "id": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
  "identity": Object {
    "accountId": "",
    "apiKey": "",
    "caller": "",
    "cognitoAuthenticationProvider": "",
    "cognitoAuthenticationType": "",
    "cognitoIdentityId": "",
    "cognitoIdentityPoolId": "",
    "sourceIp": "11.111.111.111",
    "user": "",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36",
    "userArn": "",
  },
  "method": "GET",
  "og": Object {
    "context": undefined,
    "event": Object {
      "body": "{\\"name\\": \\"Ethan\\"}",
      "headers": Object {
        "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8",
        "Accept-Encoding": "gzip, deflate, sdch, br",
        "Accept-Language": "en-US,en;q=0.8",
        "CloudFront-Forwarded-Proto": "https",
        "CloudFront-Is-Desktop-Viewer": "true",
        "CloudFront-Is-Mobile-Viewer": "false",
        "CloudFront-Is-SmartTV-Viewer": "false",
        "CloudFront-Is-Tablet-Viewer": "false",
        "CloudFront-Viewer-Country": "US",
        "Content-Type": "application/json",
        "Host": "xxxxxxxxxx.execute-api.us-east-1.amazonaws.com",
        "Upgrade-Insecure-Requests": "1",
        "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36",
        "Via": "1.1 xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.cloudfront.net (CloudFront)",
        "X-Amz-Cf-Id": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx_xxxxxxxxxxx_xxxx==",
        "X-Forwarded-For": "11.111.111.111, 11.111.111.111",
        "X-Forwarded-Port": "111",
        "X-Forwarded-Proto": "http",
      },
      "httpMethod": "GET",
      "path": "/",
      "pathParameters": Object {
        "id": "1",
      },
      "queryStringParameters": Object {},
      "requestContext": Object {
        "accountId": "111111111111",
        "apiId": "xxxxxxxxxx",
        "httpMethod": "GET",
        "identity": Object {
          "accountId": "",
          "apiKey": "",
          "caller": "",
          "cognitoAuthenticationProvider": "",
          "cognitoAuthenticationType": "",
          "cognitoIdentityId": "",
          "cognitoIdentityPoolId": "",
          "sourceIp": "11.111.111.111",
          "user": "",
          "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36",
          "userArn": "",
        },
        "requestId": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
        "resourceId": "xxxxxx",
        "resourcePath": "/{proxy+}",
        "stage": "prod",
      },
      "resource": "/{proxy+}",
    },
  },
  "params": Object {
    "id": "1",
  },
  "provider": "aws",
  "query": Object {},
  "resource": "/{proxy+}",
  "stage": "prod",
}
`;
